tsdb/index/index.go

```diff
@@ -716,19 +724,46 @@ func newReader(b ByteSlice, c io.Closer) (*Reader, error) {
 		return nil, errors.Wrap(err, "read label index table")
 	}
 
-	r.postings[""] = map[string]uint64{}
-	if err := ReadOffsetTable(r.b, toc.PostingsTable, func(key []string, off uint64) error {
+	var lastKey []string
+	lastOff := 0
+	valueCount := 0
+	// For the postings offset table we keep every label name but only every nth
+	// label value (plus the last one), to save memory.
```

Comment:
Plus the first one! (: 
```suggestion
	// label value (plus the first and last one), to save memory.
```

Reply:
.. as the offsets are sorted by label names and values.

Reply:
Done